h2o-py/h2o/estimators/aggregator.py

```diff
@@ -366,6 +366,21 @@ def export_checkpoints_dir(self, export_checkpoints_dir):
         self._parms["export_checkpoints_dir"] = export_checkpoints_dir
 
 
+    @property
+    def te_model_id(self):
+        """
+        Key of TargetEncoderModel
+
+        Type: ``str``.
+        """
+        return self._parms.get("te_model_id")
+
+    @te_model_id.setter
+    def te_model_id(self, te_model_id):
+        assert_is_type(te_model_id, None, str, H2OEstimator)
+        self._parms["te_model_id"] = te_model_id
```

Comment:
if you need a more specific validation here, there's a hook in the code generator for this.
in `bin/custom/python/gen_defaults.py`, you can add:
```
overrides=dict(
    te_model_id=dict(
        setter="""
assert_is_type(te_model_id, None, str, H2OTargetEncoderEstimator)
self._parms["te_model_id"] = te_model_id.key if isinstance(te_model_id, H2OEstimator) else te_model_id
""")
)
```

the same should have been done for `checkpoints` instead of the hardcoded logic in the abstract `estimator_base`.

Reply:
@sebhrusen My understanding is that ModelKeyV3 on backend results in generated client's code for  assertion only against H2OEstimator. As you said, for more specific validation ( against  H2OTargetEncoderEstimator let say) we can use this hook that you mentioned. I can't understand a couple of things:
1) I tried to add snippet to `gen_defaults.py` but execution of `gen_python.py` would not generate any new code. Shouldn't it? I was not sure about a proper syntax in `gen_defaults.py` and just made some changes in existing string literals.... rerun generator and nothing changed. What am I missing here?
2) What would be an alternative for R side? Do we have hooks there?
3) Actually I don't see  even more generic validation to be working when I'm passing dummy string instead of key:
```
H2OGradientBoostingEstimator(...., te_model_id = "absdefg")

@te_model_id.setter
    def te_model_id(self, te_model_id):
        assert_is_type(te_model_id, None, str, H2OEstimator)
```
is not failing for me.

Reply:
@deil87 I had to apply a small fix to `gen_python.py`: do you want me to push it your branch?
For R, there's no equivalent unfortunately.

Reply:
or I can give you a patch

Reply:
@sebhrusen I will take a patch :)  if there is no equivalent for R, probably it is better to leave python version without a hook as well.